### DISCUSSION TITLE
What are the differences between Rye and Pixi?

### DISCUSSION THREAD

**Discussion by linhduongtuan** [astral-sh/rye#419] [Q&A]
As far as I know, Rye and Pixi are quite similar?

**Comment by 3f6a** [comment]
To give some context, `pixi` is a new package manager for Python, for the conda ecosystem. Version 0.1.0 was just launched a couple of days ago, see https://prefix.dev/blog/launching_pixi. It looks quite nice.

So far I think they do not support `pip` packages yet, but it's in their pipeline to do so.

I would also be curious to know people's opinion on it, and how it compares to `rye`.

**Reply by 3f6a** [reply]
See also: https://github.com/prefix-dev/pixi/issues/260

**Comment by jdumas** [comment]
On this topic, I also found out about [huak](https://github.com/cnpryer/huak), which is another Python package manager written in Rust. I don't know how mature it is. It seems to behave similarly to Hatch/PDM/Rye, but doesn't manage the Python toolchain itself.

**Comment by gokulyc** [comment]
there is a package for Pypi packages from pixi.

https://github.com/prefix-dev/rip

is there any plan to use rip instead of pip-tools?

**Comment by majidaldo** [comment]
They are not similar. Pixi is "higher-level" and uses the conda ecosystem to deal with packages in a language-agnostic way. Rye is just to manage Python projects.

Potentially, you could use them together.

**Comment by philippemiron** [comment]
I guess one of the main differences is that pixi is conda based and uses a custom `pixi.toml` to manage the environment. Rye follows PEP standard (uses venv, pip, pip-tools under the hood) and `pyproject.toml` with standard tags (exception of it's own tool section).

**Comment by 3f6a** [comment]
I guess it would be nice to have some sort of "official" note explaining the differences between the two tools, from an end-user standpoint. Like perhaps a blogpost or something. So far for me both have been kind of equivalent, though sometimes `rye` seems to behave better so I tend to use it more (though I don't really understand why).

By "official" I mean coming from the developers of the tools, since probably they understand better their roadmap, current and future features, where the tools are similar/different today, and in the future.

Generally, some guidance in choosing one or the other would be nice.

**Comment by gregorywaynepower** [comment]
Anybody in the future who is looking for a comparison:

- Pixi uses uv for PyPI dependency management.
- Pixi is useful for managing  packages/dependencies outside the Python ecosystem.
